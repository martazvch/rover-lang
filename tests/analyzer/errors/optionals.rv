fn main() {
    var i = 0

    if i == null {}
    if null != i {}

    var b: ?bool = null

    if b == 1 {}
    if 1 != b {}

    if b {}
    if not b {}

    // Not allowed
    var c: float = null

    var d: ?float = 5.6
    print d + 65.1
}
