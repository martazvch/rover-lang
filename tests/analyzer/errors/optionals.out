invalid_comparison, int, null
invalid_comparison, null, int

non_null_comp_optional, int
non_null_comp_optional, int

non_bool_cond, if, ?bool
invalid_unary, ?bool

null_assign_to_non_optional, float
invalid_arithmetic, ?float
